src/query/WhereQuery.ts

```diff
@@ -1,7 +1,18 @@
-import type { Entity, EntityFieldValue } from '../Entity';
+import type { EntityFieldValue } from '../Entity';
 
-export type WhereClauseValue = Entity | EntityFieldValue | WhereQuery;
+export type WhereClauseValue = EntityFieldValue | WhereQuery;
 
 export interface WhereQuery {
   [index: string]: WhereClauseValue;
 }
+
+export type WhereQueryTyped<T> = {
+  [P in keyof T]?: WhereClauseValue;
+} & {
+  where?: WhereQueryTyped<T>;
```

Comment:
I don't think this is valid for a where query.

Reply:
I saw this syntax being used in platform on the ```count(where: WhereQuery)``` functions. Was the type of the count function supposed to be ```WhereQueryTyped<T> | FindArgsTyped<T>```?